Development/ssl/ssl_utils.cpp

```diff
@@ -0,0 +1,324 @@
+#include "ssl/ssl_utils.h"
+
+#include <boost/algorithm/string.hpp> // for boost::split
+#include <openssl/err.h>
+#include <openssl/pem.h>
+
+namespace ssl
+{
+    namespace experimental
+    {
+        namespace details
+        {
+            // get the Relative Distinguished Name Sequence in the format specified in RFC2253
+            // see https://www.rfc-editor.org/rfc/rfc2253#section-2
+            inline std::string get_relative_distinguished_name_sequence(X509_NAME* x509_name)
+            {
+                std::string result;
+
+                BIO_ptr bio(BIO_new(BIO_s_mem()), &BIO_free);
+                if (!bio)
+                {
+                    throw ssl_exception("failed to get_relative_distinguished_name_sequence while creating BIO to load OCSP request: BIO_new failure: " + last_openssl_error());
+                }
+                X509_NAME_print_ex(bio.get(), x509_name, 0, XN_FLAG_RFC2253);
+                char buffer[128];
+                int bytes_read;
+                while ((bytes_read = BIO_gets(bio.get(), &buffer[0], sizeof(buffer))) > 0)
+                {
+                    result.append(buffer);
+                }
+                return result;
+            }
+
+            // get the Relative Distinguished Names from the given X509 Distinguished Name
+            // see https://www.rfc-editor.org/rfc/rfc2253#section-2.1
+            inline std::vector<std::string> get_relative_distinguished_names(X509_NAME* x509_name)
+            {
+                const auto relative_distinguished_name_sequence = get_relative_distinguished_name_sequence(x509_name);
+
+                std::vector<std::string> relative_distinguished_names;
+                boost::split(relative_distinguished_names, relative_distinguished_name_sequence, boost::is_any_of(","));
```

Comment:
I think this needs to be more complicated to handle escaped characters, e.g. `\,`, see https://tools.ietf.org/html/rfc2253#section-5

Reply:
Fixed in the latest check-in